packages/iTwinUI-react/src/core/utils/components/MiddleTextTruncation.tsx

```diff
@@ -19,33 +19,37 @@ export type MiddleTextTruncationProps = {
    */
   endCharsCount?: number;
   /**
-   * Callback fired when text is truncated.
+   * Custom renderer for the truncated text.
    */
-  onTruncate?: (isTruncated: boolean) => void;
+  textRenderer?: (
+    truncatedText: string,
+    originalText: string,
```

Comment:
What is the reason you are passing `originalText`? User would have it as he is passing it to us

Reply:
In general, I think that the user will try to reuse the text rendering functions in multiple places, but after truncation, the text may become heavily modified beyond recognition, so it will lose the reference to the source. Thus, the user won't be able, for example, to show a tooltip with original text.